### PR TITLE
BaSP-TG-02: get, get by id, create endpoints for employees

### PR DESCRIPTION
**Pull request by JuliBalcaza**

### CODE REVIEW

File: src/controllers/employees.js [hunk 1]
```diff
@@ -1,129 +1,64 @@
...
-    employees.push(saveEmployee);
-    fs.writeFile('./src/data/employees.json', JSON.stringify(employees), (err) => {
-      if (err) {
-        res.status(500).json({ error: 'Error writing file' });
-      } else {
-        res.status(201).json({ newEmployee });
-      }
+const getAllEmployees = async (req, res) => {
+  try {
+    const employees = await Employees.find();
```
Comment (lmalvasia) [inline src/controllers/employees.js:5]: use query params here to filter the data.
```suggestion
    const employees = await Employees.find(req.query);
```

File: src/controllers/employees.js [hunk 2]
```diff
@@ -1,129 +1,64 @@
...
-        employee.email = newEmployee.email ? newEmployee.email : employee.email;
-        employee.password = newEmployee.password ? newEmployee.password : employee.password;
-        employee.name = newEmployee.name ? newEmployee.name : employee.name;
-        employee.lastName = newEmployee.lastName ? newEmployee.lastName : employee.lastName;
-        employee.phone = newEmployee.phone ? newEmployee.phone : employee.phone;
-      }
+const getEmployeesById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const employee = await Employees.findById(id);
```
Comment (lmalvasia) [inline src/controllers/employees.js:22]: the `findById` method could return an undefined/null value, so you have to add an extra validation to check this and to return a different response based on the value of the `employee` variable.

File: src/controllers/employees.js [hunk 3]
```diff
@@ -1,129 +1,64 @@
...
+  } catch (error) {
+    return res.status(400).json({
+      message: ' Something went wrong',
+      error: true,
+    });
+  }
 };
 
-module.exports = {
+export default {
```
Comment (lmalvasia) [inline src/controllers/employees.js:60]: Don't use export default here. You have to export each function individually.

File: src/index.js [hunk 4]
```diff
@@ -1,12 +1,16 @@
 // use "import" to import libraries
 import express from 'express';
 import mongoose from 'mongoose';
+import employeesRoutes from './routes/employees';
 
 const app = express();
-const port = 5000;
+const port = 3000;
```
Comment (lmalvasia) [inline src/index.js:7]: Why you change this? We have to use always the same port

File: src/index.js [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 
 app.use(express.json());
 
+// Entity Routes
+app.use('/employees', employeesRoutes);
```
Comment (lmalvasia) [inline src/index.js:12]: You have to use the router exported from `src/routes/index.js`. Inside that file you have to use add this line. (Check the PR from one of your teammates).

File: src/index.js [hunk 6]
```diff
@@ -1,12 +1,16 @@
...
 import mongoose from 'mongoose';
+import employeesRoutes from './routes/employees';
 
 const app = express();
-const port = 5000;
+const port = 3000;
 
 app.use(express.json());
 
+// Entity Routes
```
Comment (lmalvasia) [inline src/index.js:11]: Remove this comment

File: src/validations/employees.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import Joi from 'joi';
+
+const validateEmployeesCreation = (req, res, next) => {
```
Comment (lmalvasia) [inline src/validations/employees.js:3]:
```suggestion
const validateEmployeeBody = (req, res, next) => {
```

File: src/validations/employees.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    return res.status(400).json({
+      message: `Something went wrong: ${validation.error.details[0].message} `,
+      data: undefined,
+      error: true,
+    });
+  }
+  return next();
+};
+
+export default {
```
Comment (lmalvasia) [inline src/validations/employees.js:24]: export default validateEmployeeBody;

File: src/index.js (hunk 4 above)
Comment (JuliBalcaza) [reply src/index.js:7]: I had to change it because at first, port 5000 didn't work for me, but now it does, I'll change it again.

**Review by kolodani — APPROVED** [review]
great work

**Pull request merged**